addons/website/static/src/interactions/video/media_video.js

```diff
@@ -0,0 +1,98 @@
+import { Interaction } from "@website/core/interaction";
+import { registry } from "@web/core/registry";
+
+import { _t } from "@web/core/l10n/translation";
+import { escape } from "@web/core/utils/strings";
+
+import { setupAutoplay, triggerAutoplay } from "../../core/video_utils";
+
+class MediaVideo extends Interaction {
+    static selector = ".media_iframe_video";
+
+    setup() {
+        if (this.el.dataset.needCookiesApproval) {
+            this.sizeContainerEl = this.el.querySelector(":scope > .media_iframe_video_size");
+            this.sizeContainerEl.classList.add("d-none");
+            this.addListener(document, "optionalCookiesAccepted", this.sizeContainerEl.classList.remove("d-none"))
+            this.registerCleanup(() => this.sizeContainerEl.classList.remove("d-none"));
+        }
+    }
+
+    start() {
+        let iframeEl = this.el.querySelector(':scope > iframe');
+
+        // The following code is only there to ensure compatibility with
+        // videos added before bug fixes or new Odoo versions where the
+        // <iframe/> element is properly saved.
+        if (!iframeEl) {
+            iframeEl = this.generateIframe();
+        }
+
+        // We don't want to cause an error that would prevent entering edit mode
+        // if there is an iframe that doesn't have a src (this was possible for
+        // a while with the media dialog).
+        if (!iframeEl || !iframeEl.getAttribute('src')) {
```

Comment:
iframeEl is guaranteed to exist here i believe, so you can just skip the first condition i guess

Reply:
`generateIframe` does not return an Iframe if the source is not valid. Therefore the first condition is necessary (or at least we can use `!iframeEl?.getAttribute("src")`